File: wanted-cqrs/src/main/java/com/mkhwang/wantedcqrs/product/domain/dto/ProductOptionGroupDto.java
package com.mkhwang.wantedcqrs.product.domain.dto;

import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
public class ProductOptionGroupDto {
  private Long id;
  private String name;
  private int displayOrder;
  private List<ProductOptionDto> options;
}

File: src/product/application/dto/ProductOptionGroup.dto.ts
import Product_Option from "@product/domain/entities/Product_Option";

export default class ProductOptionGroupDTO {
  constructor(
    public name: string,
    public display_order: number,
    public options?: Omit<Product_Option, "option_group_id">[],
    public product_id?: number,
    public id?: number,
  ) {}
}
